### DISCUSSION TITLE
Parse Triangular mesh to Tetrahedral Volume Mesh in order to execute BBW

### DISCUSSION THREAD

**Discussion by mobereng** [libigl/libigl#1760] [General]
Is there a way to parse from a common triagular mesh to a tetrahedral volume one ?

**Comment by jdumas** [comment]
What do you mean? If you want to read a tet mesh, libigl supports .msh file format.

**Reply by mobereng** [reply]
No, let me explain a bit more

Currently I have an skeletonization algorithm implemented, but it was done only with a triangular mesh (V, F), now I'm trying to use that data, which is the V and F matrices and the skeleton as TGF file., to apply BBW algorithm. So basically I have everything but my model is not a tet

Knowing that, I experiment a bit with TetGen library but with no success, I used the default recommended flags `"pq1.414a0.01"` for `igl::copyleft::tetgen::tetrahedralize` in the following manner `igl::copyleft::tetgen::tetrahedralize(OV, OF, "pq1.414a0.01", V, T, F);`

So as you can see my goal is to achieve a tetrahedralization compatible with BBW

**Reply by jdumas** [reply]
What's wrong with tetgen?

**Reply by mobereng** [reply]
At the moment the error is not tetgen itself, but the tet which is being generated after `tetrahedralize` at the moment it creates the tet but as soon as I start the BBWs calculation it fails

![image](https://user-images.githubusercontent.com/69315094/111833063-fb3ed300-88c7-11eb-985f-a9f9bcf1b867.png)
